pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -936,12 +936,12 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 				{
 					CertificateChain: &core.DataSource{
 						Specifier: &core.DataSource_Filename{
-							Filename: tls.ClientCertificate,
+							Filename: model.GetOrDefaultFromMap(metadata, model.NodeMetadataTLSClientCertChain, tls.ClientCertificate),
```

Comment:
This feels odd to me that when `tls.mode: MUTUAL` is set in `DestinationRule`, we skip over other tls.<params> provided along with it and instead use node metadata as default. i.e., priority to node metadata instead of a custom override provided by user. I feel it should be other way around, like if `tls.ClientCertificate` is not provided, we use node metadata as default.

Reply:
"priority to node metadata instead of a custom override provided by user." The node metadata *is* the custom override provided by the user. The destinationrule contains the default cluster-wide settings.